src/common/buffer.cc

```diff
@@ -125,10 +128,12 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     if (r < 0)
       return r;
     buf[r] = '\0';
-    size_t size = strict_strtol(buf, 10, &err);
-    if (!err.empty())
-      return -EIO;
-    buffer_max_pipe_size.set(size);
+    errno = 0;
+    size_t size = strtoul(buf, nullptr, 10);
+    if (errno == 0)
```

Comment:
@aclamk see http://pubs.opengroup.org/onlinepubs/9699919799/functions/strtol.html.

can you might want to try following code:
```c++
#include <iostream>
#include <cstdlib>

using namespace std;

int main()
{
  const char *s = "hi";
  char *end;
  long long ret = strtoll(s, &end, 10);
  cout << ret << endl;
  cout << (end == s) << endl;
  cout << errno << endl;

}
```